agent/cmd/cmd.go

```diff
@@ -100,14 +110,42 @@ func RunWithConfig(flags *Flags, config Config) {
 		panic("must specify non-zero agent registry port")
 	}
 
-	zlog := log.ConfigureLogger(config.ZapLogging)
-	defer zlog.Sync()
+	var overrides options
+	for _, o := range opts {
+		o(&overrides)
+	}
 
-	stats, closer, err := metrics.New(config.Metrics, flags.KrakenCluster)
-	if err != nil {
-		log.Fatalf("Failed to init metrics: %s", err)
+	var config Config
+	if overrides.config != nil {
+		config = *overrides.config
+	} else {
+		if err := configutil.Load(flags.ConfigFile, &config); err != nil {
+			panic(err)
+		}
+		if flags.SecretsFile != "" {
+			if err := configutil.Load(flags.SecretsFile, &config); err != nil {
+				panic(err)
+			}
+		}
+	}
+	config = config.applyDefaults()
+
+	if overrides.logger != nil {
+		log.SetGlobalLogger(overrides.logger.Sugar())
+	} else {
+		zlog := log.ConfigureLogger(config.ZapLogging)
+		defer zlog.Sync()
+	}
+
+	stats := overrides.metrics
+	if stats == nil {
+		s, closer, err := metrics.New(config.Metrics, flags.KrakenCluster)
```

Comment:
can we not set this to `stats` here directly? or are we worried the `err` can be non nil and `s` will also not be `nil`? quite possible it's best to leave this as is, just wanted to draw your attention to it in case it's not actually necessary

Reply:
The alternative is:
```
var closer io.Closer
var err error
stats, closer, err = metrics.New(...)
...
```

It's just more verbose, hence `stats = s`.